azurerm/resource_arm_api_management_logger.go

```diff
@@ -0,0 +1,256 @@
+		}
+	}
+
+	return nil
+}
+
+func resourceArmApiManagementLoggerUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).apiManagementLoggerClient
+	ctx := meta.(*ArmClient).StopContext
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	serviceName := d.Get("api_management_name").(string)
+	name := d.Get("name").(string)
+
+	eventHubRaw, hasEventHub := d.GetOk("eventhub")
+	appInsightsRaw, hasAppInsights := d.GetOk("application_insights")
+
+	parameters := apimanagement.LoggerUpdateContract{
+		LoggerUpdateParameters: &apimanagement.LoggerUpdateParameters{
+			IsBuffered:  utils.Bool(d.Get("buffered").(bool)),
+			Description: utils.String(d.Get("description").(string)),
+		},
+	}
+
+	if hasEventHub {
+		parameters.LoggerType = apimanagement.AzureEventHub
+		parameters.Credentials = expandArmApiManagementLoggerEventHub(eventHubRaw.([]interface{}))
+	} else if hasAppInsights {
+		parameters.LoggerType = apimanagement.ApplicationInsights
+		parameters.Credentials = expandArmApiManagementLoggerApplicationInsights(appInsightsRaw.([]interface{}))
+	}
+
+	if _, err := client.Update(ctx, resourceGroup, serviceName, name, parameters, ""); err != nil {
+		return fmt.Errorf("Error updating Logger %q (Resource Group %q / API Management Service %q): %+v", name, resourceGroup, serviceName, err)
+	}
+
+	return resourceArmApiManagementLoggerRead(d, meta)
+}
+
+func resourceArmApiManagementLoggerDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).apiManagementLoggerClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return fmt.Errorf("Error parsing API Management Logger ID %q: %+v", d.Id(), err)
+	}
+	resourceGroup := id.ResourceGroup
+	serviceName := id.Path["service"]
+	name := id.Path["loggers"]
+
+	if resp, err := client.Delete(ctx, resourceGroup, serviceName, name, ""); err != nil {
+		if !utils.ResponseWasNotFound(resp) {
+			return fmt.Errorf("Error deleting Logger %q (Resource Group %q / API Management Service %q): %+v", name, resourceGroup, serviceName, err)
+		}
+	}
+
+	return nil
+}
+
+func expandArmApiManagementLoggerEventHub(input []interface{}) map[string]*string {
+	credentials := make(map[string]*string)
+	eventHub := input[0].(map[string]interface{})
+	credentials["name"] = utils.String(eventHub["name"].(string))
+	credentials["connectionString"] = utils.String(eventHub["connection_string"].(string))
+	return credentials
+}
+
+func expandArmApiManagementLoggerApplicationInsights(input []interface{}) map[string]*string {
+	credentials := make(map[string]*string)
+	ai := input[0].(map[string]interface{})
+	credentials["instrumentationKey"] = utils.String(ai["instrumentation_key"].(string))
+	return credentials
+}
+
+func flattenArmApiManagementLoggerEventHub(d *schema.ResourceData, properties *apimanagement.LoggerContractProperties) []interface{} {
+	if properties.LoggerType != apimanagement.AzureEventHub {
```

Comment:
we should probably set this in all circumstances; since Terraform's state should match the resource's state in Azure